### ISSUE TITLE
disable schema1 manifest pushes by default + option to enable

### ISSUE THREAD

**Issue by vikstrous** [docker/distribution#2472]
Proposal:

1. Add a new error `ErrSchemaV1Unsupported`
2. Add a new config: `compatibility.schema1.enabled` that defaults to false

I think the way I'll go about implementing this is:

1. Create a new `ManifestHandler` that always returns ErrSchemaV1Disabled on Put and delegates to the schema1 handler on Unmarshal
2. During startup, in `docker/distribution/registry/storage/registry.go`, if the config is not "enabled", instaed of using `signedManifestHandler` as the handler for schema 1, I'll wire in the error handler

Basically, this will prevent new schema1 manifests from being pushed, but existing ones will continue to be served.

**Comment by stevvooe** [comment]
Small nit: `ErrSchemaV1Disabled` -> `ErrManifestUnsupported`.

How will this integrate with mediatypes, in general? Can you disable other mediatypes or is schema1 special? (It probably is)

Assuming that we can still serve schema1 to old clients but this will disable accepting them as the primary media type.

**Comment by vikstrous** [comment]
Proposal updated. I was planning to change only the handler, so nothing about the media types mapping to handlers would change. I think the empty media type is interpreted as schema1 and this wouldn't change, so the media type would probably become mandatory.
